tests/containers/podman_netavark.pm

```diff
@@ -113,17 +127,19 @@ sub run {
     assert_script_run("podman run --network $net2->{name}:ip=$ctr2->{ip},mac=$ctr2->{mac} --network $net1->{name}:ip=$ctr2->{ip_sec},mac=$ctr2->{mac_sec} -dt --name $ctr2->{name} $ctr2->{image}");
 
     # second container should have 2 interfaces
-    my $net1_reg = qr@ether\s+$ctr2->{mac}.*\s+inet\s+$ctr2->{ip}\/16@;
-    my $net2_reg = qr@ether\s+$ctr2->{mac_sec}.*\s+inet\s+$ctr2->{ip_sec}\/16@;
-    validate_script_output("podman exec -t $ctr2->{name} /bin/sh -c 'ip addr show eth0'", sub { /$net1_reg|$net2_reg/m });
-    validate_script_output("podman exec -t $ctr2->{name} /bin/sh -c 'ip addr show eth1'", sub { /$net1_reg|$net2_reg/m });
-
-    # busybox container should be able to resolve apache container
-    validate_script_output("podman exec -t $ctr2->{name} /bin/sh -c 'nslookup $ctr1->{name}'", sub { /Name:\s+$ctr1->{name}\s+Address.*$ctr1->{ip}/m });
-    validate_script_output("podman exec -t $ctr2->{name} /bin/sh -c 'wget -S $ctr1->{ip}:80'", sub { /HTTP.* 200 OK/ });
-    assert_script_run("grep $ctr1->{name} /run/containers/networks/aardvark-dns/$net1->{name}");
-    assert_script_run("grep $ctr2->{name} /run/containers/networks/aardvark-dns/$net1->{name}");
-    assert_script_run("grep $ctr2->{name} /run/containers/networks/aardvark-dns/$net2->{name}");
+    if (is_container_running($ctr1->{name}, $ctr2->{name})) {
+        my $net1_reg = qr@ether\s+$ctr2->{mac}.*\s+inet\s+$ctr2->{ip}\/16@;
+        my $net2_reg = qr@ether\s+$ctr2->{mac_sec}.*\s+inet\s+$ctr2->{ip_sec}\/16@;
+        validate_script_output("podman exec -t $ctr2->{name} /bin/sh -c 'ip addr show eth0'", sub { /$net1_reg|$net2_reg/m });
+        validate_script_output("podman exec -t $ctr2->{name} /bin/sh -c 'ip addr show eth1'", sub { /$net1_reg|$net2_reg/m });
+        validate_script_output("podman exec -t $ctr2->{name} /bin/sh -c 'nslookup $ctr1->{name}'", sub { /Name:\s+$ctr1->{name}\s+Address.*$ctr1->{ip}/m });
+        validate_script_output("podman exec -t $ctr2->{name} /bin/sh -c 'wget -S $ctr1->{ip}:80'", sub { /HTTP.* 200 OK/ });
+        # busybox container should be able to resolve apache container
+        assert_script_run("grep $ctr1->{name} /run/containers/networks/aardvark-dns/$net1->{name}");
+        assert_script_run("grep $ctr2->{name} /run/containers/networks/aardvark-dns/$net1->{name}");
```

Comment:
Is this correct? I would assume this would fail due to the different interface names, no?

Reply:
No, both of the containers should be in the same network segment

Reply:
Ah ok, thnx!